pulsar-client-tools/src/main/java/com/yahoo/pulsar/admin/cli/PulsarAdminTool.java

```diff
@@ -108,8 +121,28 @@ boolean run(String[] args) {
             String cmd = jcommander.getParsedCommand();
             JCommander obj = jcommander.getCommands().get(cmd);
             CmdBase cmdObj = (CmdBase) obj.getObjects().get(0);
+            if (StringUtils.isNotBlank(serviceUrl) || (StringUtils.isNotBlank(authPluginClassName) && StringUtils
+                    .isNotBlank(authParams))) {
+                if (StringUtils.isNotBlank(serviceUrl)) {
+                    try {
+                        url = new URL(serviceUrl);
+                    } catch (MalformedURLException e) {
+                        System.err.println("Invalid serviceUrl: '" + serviceUrl + "'");
+                        System.exit(1);
+                    }
+                }
+                try {
+                    if (StringUtils.isNotBlank(authPluginClassName) && StringUtils.isNotBlank(authParams)) {
+                        config.setAuthentication(authPluginClassName, authParams);
+                    }
+                    cmdObj = cmdObj.getClass().getConstructor(PulsarAdmin.class).newInstance(new PulsarAdmin(url, config));
```

Comment:
Yes, I think it's better to not change cmd-processing logic in `run()` but instead handle parsing at `main()` method which should prepare `ClientConfiguration` and calls `PulsarAdminTool(URL url, ClientConfiguration config)`